### DISCUSSION TITLE
Is glesv2 a mandatory dependency on Raspberry Pi ???

### DISCUSSION THREAD

**Discussion by WebReflection** [Igalia/cog#611] [Q&A]
I've just tried again `cog` and I always get a Display EGL failure at bootstrap, but only when running it via Weston on a Raspberry Pi. The whole thing looks half accelerated too, where I have little FPS for [the classic jellyfish demo](https://archibold.io/demo/jellyfish/) while I used to be able to go at nearly stable 60FPS before.

The only thing I am not using though is the explicit cairo-glesv2 version of the library, but I thought with latest mesa and VideoCore mainstream support that wouldn't be needed ... am I wrong in assuming this?

Also, I have cairo-glesv3 too available, but I haven't seen that mentioned in README ... is GLES v3 any better or nothing really works or changes if I use that instead? (I could try, but hearing from you would be a way better outcome).

Thank you!

**Comment by WebReflection** [comment]
FYI even using those glesv2 or glesv3 solutions I can't avoid the warning about the Display https://github.com/Igalia/cog/issues/614

**Reply by clopez** [reply]
Please check your CPU governor and ensure you don't have set it up to `powersave`. Powersave governor under-clocks the CPU.

I have recently discovered an [issue with the default configuration for a RPi build with Yocto where the CPU governor is set to powersave](https://github.com/agherzan/meta-raspberrypi/pull/1231).

Setting the governor to `performance` instead is recommended. [See here how to change it](https://github.com/Igalia/meta-webkit/wiki/PerformanceTips#set-cpu-governor-to-performance).

**Reply by WebReflection** [reply]
I will check that, thanks, although my question around `EGLDisplay Initialization failed: EGL_NOT_INITIALIZED` doesn't seem to be answered :thinking:

**Reply by clopez** [reply]
That warning can be ignored, things usually work despite of it.

**Reply by WebReflection** [reply]
OK, I can ignore the warning, but it used to go around 50FPS on a Pi Zero 2W https://www.youtube.com/watch?v=kkqheU0PnjI

now on a Pi4 it never goes faster than 18FPS, it doesn't even matter if I am running at 720p or 1080p, it's basically "as slow" as Firefox with WebGL but it used to be much faster.

I've seen many things changed here and there:

  * the `weston --socket=wpe` does nothing now
  * the `WAYLAND_DISPLAY=wpe` breaks, likely due previous point
  * before I used xorg that was launching weston with xwayland and then cog ... now I am just using weston and cog (no startx, no xorg-server, no xwayland) and while I have no errors whatsoever, and the HW acceleration seems to work, it's really underwhelming compared to 1 year ago

Is there anyone able to run this demo with WPEWebKit and on a Raspberry Pi at more than 18FPS?
https://archibold.io/demo/jellyfish/

Thank you! I feel like I might be missing something really silly or simple, but from a whole updated stack, including better Mesa support for VC4, I can't believe I am incapable of going as fast as I used to when things were a bit sketchy on all fronts!

**Reply by clopez** [reply]
Have you checked which CPU governor is using your RPi?
Please execute this command:
`cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor`
If it says `powersave` then that is your issue. You have to change it to `ondemand` or `performance`

**Reply by WebReflection** [reply]
I've used cpupower with performance governor and literally nothing changed ... I will try to see if that other command makes any difference there, thanks.

I still would like to have anyone testing that page on a Pi4 and tell me what is the FPS count so we are sure it's my issue to solve.

**Reply by WebReflection** [reply]
just checked, it says performance ... it'd be great if anyone with a Pi4 could confirm the FPS count in this demo: https://archibold.io/demo/jellyfish/
